reboot_required/check.py

```diff
@@ -0,0 +1,61 @@
+# vim: ts=4:sw=4:et
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# project
+from checks import AgentCheck
+
+from os import stat, utime, remove
+from os.path import isfile
+from stat import ST_MTIME
+from datetime import datetime, timedelta
+
+class RebootRequiredCheck(AgentCheck):
+
+    REBOOT_SIGNAL_FILE = '/var/run/reboot-required'
+    CREATED_AT_FILE = '/var/run/reboot-required.created_at'
+
+    def check(self, instance):
+        status, days_since, msg = self._check(instance)
+        self.service_check('system.reboot_required', status, message=msg)
+
+    def _check(self, instance):
+
+        reboot_signal_file = instance.get('reboot_signal_file', self.REBOOT_SIGNAL_FILE)
+        created_at_file = instance.get('created_at_file', self.CREATED_AT_FILE)
+        warning_days = int(instance.get('days_warning', 7))
+        critical_days = int(instance.get('days_critical', 14))
+
+        return self._get_status(critical_days, warning_days, self._days_since(reboot_signal_file, created_at_file))
+
+    def _days_since(self, reboot_signal_file, created_at_file):
+        if isfile(reboot_signal_file):
+            if isfile(created_at_file):
+                created_at = self._get_created_at(created_at_file)
```

Comment:
Why create the `reboot-required.created_at` file and not directly stat `/var/run/reboot-required` ?

Reply:
reboot-required is frequently updated as packages are installed, as I understand it, stat will only ever return the last modified time on many systems (windows being the exception). Since what I really want to track is the creation time of this file since the last reboot I decided I would tackle tracking that point in time this way. There are other tricks one can use but this seemed the easiest.

Reply:
Make sense, I did not think about multiple updates.